docs/testing-overview.md

```diff
@@ -16,6 +31,195 @@ Code style in JavaScript is enforced using [ESLint](http://eslint.org/). The abo
 
 To run unit tests only, without the linter, use `npm run test-unit` instead.
 
+## Writing tests
+
+We use [Jest](https://facebook.github.io/jest) testing framework for unit tests, and [Enzyme](https://github.com/airbnb/enzyme) for testing React components. You can find more information on testing components at [component tests](component-tests.md).
+
+Though you'll still see Chai assertions and Sinon spies/mocks throughout the code base, for all new tests we use the Jest API for [globals](https://facebook.github.io/jest/docs/en/api.html) (`describe`, `test`, `beforeEach` and so on) [assertions](http://facebook.github.io/jest/docs/en/expect.html), [mocks](http://facebook.github.io/jest/docs/en/mock-functions.html), [spies](http://facebook.github.io/jest/docs/en/jest-object.html#jestspyonobject-methodname) and [mock functions](https://facebook.github.io/jest/docs/en/mock-function-api.html).
+
+## Folder structure
+Keep your tests in a `test` folder in your working directory. The test file should have the same name as the test subject file.
+
+```
++-- test
+|   +-- bar.js
++-- bar.jsx
+```
+
+Only test files (with at least one test case) should live directly under `/test`. If you need to add external mocks or fixtures, place them in a sub folder, for example:
+
+* `test/mocks/[file-name.js`
+* `test/fixtures/[file-name].js` 
+
+### Importing tests
+
+Given the previous folder structure, try to use relative paths when importing of the __code you're testing__, as opposed to using project paths.
+
+**Good**
+
+`import { bar } from '../bar';`
+
+**Not so good**
+
+`import { bar } from 'components/foo/bar';`
+
+It will make your life easier should you decide to relocate your code to another position in the application directory.
+
+### Describing tests
+Use a `describe` block to group test cases. Each test case should ideally describe one behaviour only.
+
+In test cases, try to describe in plain words the expected behaviour. For UI components, this might entail describing expected behaviour from a user perspective rather than explaining code internals.
+
+**Good**
+
+```javascript
+describe( 'CheckboxWithLabel', () => {
+    test('checking checkbox should disable the form submit button', () => {
+        ...
+    } );
+} );
+```
+
+**Not so good**
+
+```javascript
+describe( 'CheckboxWithLabel', () => {
+    test('checking checkbox should set this.state.disableButton to `true`', () => {
+        ...
+    } );
+} );
+```
+
+### Snapshot testing
```

Comment:
There's another "Snapshot testing" section, maybe we can remove one of them?

Reply:
Yes, I had the same concern. I would remove the short one that is included in this PR.